src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1988,22 +1988,31 @@ void Unit::AttackerStateUpdate(Unit* victim, WeaponAttackType attType, bool extr
         m_currentSpells[CURRENT_MELEE_SPELL]->cast();
     else
     {
-        // attack can be redirected to another target
-        victim = GetMeleeHitRedirectTarget(victim);
+        std::list<AuraEffect*> meleeAttackOverrides = GetAuraEffectsByType(SPELL_AURA_OVERRIDE_AUTOATTACK_WITH_MELEE_SPELL);
```

Comment:
use `AuraEffectList const&` instead of `std::list<AuraEffect*>` or alternatively use `std::list<AuraEffect*> const&`